pkg/api/validation/validation.go

```diff
@@ -145,6 +152,23 @@ func ValidateEndpointsSpecificAnnotations(annotations map[string]string, fldPath
 	return allErrs
 }
 
+func validateOwnerReference(ownerReference api.OwnerReference, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	gvk := unversioned.FromAPIVersionAndKind(ownerReference.APIVersion, ownerReference.Kind)
```

Comment:
I plan to let the garbage collector to compare the reference with the discovery results, and ignore the reference if the resource doesn't exist in the system.